sdk/python/feast/client.py

```diff
@@ -908,6 +987,20 @@ def _build_feature_references(
     return features
 
 
+def _generate_dataset_id(feature_set: FeatureSet) -> str:
```

Comment:
The term dataset still seems a bit off to me.

The `id` that we are returning is a `job_id` or an `ingestion_id` since it is unique to every invocatio of `ingest()`. If we refer to it as `dataset_id` then I think users would expect it to be associated with their dataset, which means they provide the value or at the very least the value doesnt change in subsequent runs.

Reply:
I was wary of letting users set this value since there is a chance they would assign the same value twice. And running the ingestion twice SHOULD have distinct ids because users probably don't want to treat the data as belonging to the same set... I can add the option to override the dataset id, but it really opens up the possibility of users breaking the system.

I don't think it should be a `job_id`, which is already an extremely overloaded term, and retrieval of data by `ingestion_id` doesn't seem as intuitive as something like a `dataset_id`

Reply:
The reason I brought this up is because the current implementation does not fit the typical 
SOP for our users where they build idempotent data system. They expect to be able to create a dataset and ingest it based on a name, and have the data system upsert. So basically they just run ingestion until it succeeds and then they are done. They don't care, and often don't know, that we are running Kafka.

And our users definitely don't want to maintain a list of UUIDs. They probably just want to query their datasets by the names they are using like "2020417-mydataset`

Regardless, I think your current approach is fine for now because nothing prevents us from eventually allowing them to set this Id if we allow idempotency. I would probably reuse this same `dataset_id` though.

Reply:
ok. I've added the option for people to supply their own dataset id, and fixed the sql so that it only retrieves the latest value for each dataset/entity/event timestamp. This should ensure idempotency.